src/app/shared/components/field-error/field-error.component.ts

```diff
@@ -9,16 +9,30 @@ import { ErrorCode, ErrorTranslationKey } from './errors';
 })
 export class FieldErrorComponent {
   @Input() model!: NgModel;
-
+  @Input() locationControl!: FormControl;
   constructor() {}
 
   // mat-error "reserves enough space to display one error message at a time" so we get first error
-  getError(): string {
-    const validationErrors = this.model.errors;
-    if (validationErrors) {
-      const errors = Object.keys(validationErrors);
-      if (errors.length > 0) {
-        return ErrorTranslationKey[errors[0] as ErrorCode];
+  getError(): any {
+    if (this.model) {
+      let validationErrors = this.model.errors;
+      if (validationErrors) {
+        const errors = Object.keys(validationErrors);
+        if (errors.length > 0) {
+          return ErrorTranslationKey[errors[0] as ErrorCode];
```

Comment:
Jaki typ zwraca `errors[0]`, że musisz robić type assertion?

Reply:
<img width="1054" alt="image" src="https://user-images.githubusercontent.com/12514322/159446000-1781c7f0-bed4-4c4e-859c-4ac12c0861fe.png">
<img width="898" alt="image" src="https://user-images.githubusercontent.com/12514322/159446149-f793ee3a-2763-4829-92e7-03cf98f57aa6.png">

Reply:
A, to pewnie przez `Object.keys`, bo typy w TS pozwalają na dodatkowe pola w wartości, których nie ma zdefiniowanych w typie, więc `Object.keys` może zagwarantować tylko, że zwraca tablicę stringów.

Rozumiem, że tu polegamy jednak na tym, że ten obiekt nie ma takich ekstra kluczy; szkoda, że nie da się tego zagwarantować w jakiś type-safe sposób.